tutorials/preprocessing/25_background_filtering.py

```diff
@@ -1,1071 +1,1071 @@
+#
+# We can recreate a problematic simulation from
+# Tanner *et al.* (2015) :footcite:`TannerEtAl2015`:
+#
+#    "The simulated component is a single-cycle cosine wave with an amplitude
+#    of 5µV [sic], onset of 500 ms poststimulus, and duration of 800 ms. The
+#    simulated component was embedded in 20 s of zero values to avoid
+#    filtering edge effects. [...] Distortions [were] caused by 2 Hz low-pass
+#    and high-pass filters. [...] No visible distortion to the original
+#    waveform [occurred] with 30 Hz low-pass and 0.01 Hz high-pass filters.
+#    [...] Filter frequencies correspond to the half-amplitude (-6 dB) cutoff
+#    (12 dB/octave roll-off)."
+#
+# .. note:: This simulated signal contains energy not just within the
+#           pass-band, but also within the transition and stop-bands -- perhaps
+#           most easily understood because the signal has a non-zero DC value,
+#           but also because it is a shifted cosine that has been
+#           *windowed* (here multiplied by a rectangular window), which
+#           makes the cosine and DC frequencies spread to other frequencies
+#           (multiplication in time is convolution in frequency, so multiplying
+#           by a rectangular window in the time domain means convolving a sinc
+#           function with the impulses at DC and the cosine frequency in the
+#           frequency domain).
+#
+
+x = np.zeros(int(2 * sfreq))
+t = np.arange(0, len(x)) / sfreq - 0.2
+onset = np.where(t >= 0.5)[0][0]
+cos_t = np.arange(0, int(sfreq * 0.8)) / sfreq
+sig = 2.5 - 2.5 * np.cos(2 * np.pi * (1. / 0.8) * cos_t)
+x[onset:onset + len(sig)] = sig
+
+iir_lp_30 = signal.iirfilter(2, 30. / sfreq, btype='lowpass')
+iir_hp_p1 = signal.iirfilter(2, 0.1 / sfreq, btype='highpass')
+iir_lp_2 = signal.iirfilter(2, 2. / sfreq, btype='lowpass')
+iir_hp_2 = signal.iirfilter(2, 2. / sfreq, btype='highpass')
+x_lp_30 = signal.filtfilt(iir_lp_30[0], iir_lp_30[1], x, padlen=0)
+x_hp_p1 = signal.filtfilt(iir_hp_p1[0], iir_hp_p1[1], x, padlen=0)
+x_lp_2 = signal.filtfilt(iir_lp_2[0], iir_lp_2[1], x, padlen=0)
+x_hp_2 = signal.filtfilt(iir_hp_2[0], iir_hp_2[1], x, padlen=0)
+
+xlim = t[[0, -1]]
+ylim = [-2, 6]
+xlabel = 'Time (s)'
+ylabel = r'Amplitude ($\mu$V)'
+tticks = [0, 0.5, 1.3, t[-1]]
+axes = plt.subplots(2, 2)[1].ravel()
+for ax, x_f, title in zip(axes, [x_lp_2, x_lp_30, x_hp_2, x_hp_p1],
+                          ['LP$_2$', 'LP$_{30}$', 'HP$_2$', 'LP$_{0.1}$']):
+    ax.plot(t, x, color='0.5')
+    ax.plot(t, x_f, color='k', linestyle='--')
+    ax.set(ylim=ylim, xlim=xlim, xticks=tticks,
+           title=title, xlabel=xlabel, ylabel=ylabel)
+mne.viz.adjust_axes(axes)
+mne.viz.tight_layout()
+plt.show()
+
+# %%
+# Similarly, in a P300 paradigm reported by
+# Kappenman & Luck (2010) :footcite:`KappenmanLuck2010`,
+# they found that applying a :unit:`1 Hz` high-pass decreased the probability of
+# finding a significant difference in the N100 response, likely because

```

Comment:
```suggestion
# they found that applying a :unit:`1 Hz` high-pass decreased the probability
# of finding a significant difference in the N100 response, likely because

```

Reply:
(max line len 79)